src/solvers/glpk.rs

```diff
@@ -118,18 +123,40 @@ impl SolverWithSolutionParsing for GlpkSolver {
     }
 }
 
+impl WithMaxSeconds<GlpkSolver> for GlpkSolver {
+    fn max_seconds(&self) -> Option<u32> {
+        self.seconds
+    }
+
+    fn with_max_seconds(&self, seconds: u32) -> GlpkSolver {
+        GlpkSolver {
+            seconds: Some(seconds),
+            ..(*self).clone()
+        }
+    }
+}
+
 impl SolverProgram for GlpkSolver {
     fn command_name(&self) -> &str {
         &self.command_name
     }
 
     fn arguments(&self, lp_file: &Path, solution_file: &Path) -> Vec<OsString> {
-        vec![
+        let mut args = vec![
             "--lp".into(),
             lp_file.into(),
             "-o".into(),
             solution_file.into(),
-        ]
+        ];
+
+        for (name, value) in [("--tmlim", self.max_seconds())].iter() {
+            if let Some(val) = value {
+                args.push(name.into());
+                args.push(val.to_string().into());
+            }
+        }
```

Comment:
Isn't that more complicated than it needs to be ?

Could we have 

```suggestion
        if let Some(seconds) = self.max_seconds() {
            args.push("--tmlim".into());
            args.push(seconds.to_string().into());
        }
```

Reply:
I copied that approach from `CbcSolver`, although that code actually uses both approaches and ends up setting the timeout arguments twice.  I figured the iterator would simplify the code as more options are added (e.g., the MIP gap option that  just proposed) but I'd be happy to change it.

Reply:
ok, I'm fine with that.